cms/admin/pageadmin.py

```diff
@@ -816,10 +782,8 @@ def changelist_view(self, request, extra_context=None):
             'tree': {
                 'site': site,
                 'sites': self.get_sites_for_user(request.user),
-                'query': query,
+                'query': request.GET.get('q', ''),
                 'is_filtered': changelist_form.is_filtered(),
-                'items': pages,
-                'has_items': has_items,
```

Comment:
Someone has gone to extra effort to keep the tree and the changelist separate. Paulo wouldn't have ran extra queries if he didn't need to.